tests/testsuite/update.rs

```diff
@@ -1740,34 +1772,113 @@ fn update_breaking() {
         .alternative(true)
         .publish();
 
-    p.cargo("update --dry-run --breaking")
-        .with_status(1)
-        .with_stderr_contains("[ERROR] unexpected argument '--breaking' found")
+    p.cargo("update -Zunstable-options --dry-run --breaking")
+        .masquerade_as_nightly_cargo(&["update-breaking"])
+        // .env("CARGO_LOG", "cargo::ops=trace")
+        .with_stderr_unordered(
```

Comment:
Is the ordering non-deterministic?  Our existing `cargo update` tests do not use `unordered`

Reply:
It is. These can come in any order:

```
Updating `dummy-registry` index
Updating `alternative` index
```

Also, this can come first or last, depending on when `bar/Cargo.toml` is upgraded.

```
Upgrading multiple ^2.0 -> v3.0.0
```

Reply:
I'm sorting the crates before upgrading, to make this deterministic. Fixed.

Reply:
Ah, the source of non-determinism is that `Workspace::members_mut` iterates over `Packages::packages` which is as `HashMap<PathBuf, Package>`.

In that case, the `sort` is likely fine